azure_functions_worker/functions.py

```diff
@@ -368,6 +378,8 @@ def add_indexed_function(self, function_id: str,
 
         bound_params = {}
         for binding in function.get_bindings():
+            self.validate_binding_route(func_name, binding, func_type)
```

Comment:
This is only useful for http. right?

If yes, then add a validate_bindings call, which internally calls validate binding route and validate binding direction in it.

In future, we might want to validate other bindings as well and we don't want to make it convulated.

Reply:
Refactored.